rkt/metadata_service.go

```diff
@@ -541,12 +546,20 @@ func runMetadataService(args []string) (exit int) {
 
 	log.Print("Metadata service running...")
 
-	if err = srv.Serve(l); err != nil {
-		log.Printf("Error serving HTTP: %v", err)
+	go func() {
+		if err = srv.Serve(unixl); err != nil {
+			stderr("Error serving registration HTTP: %v", err)
+			cleanup()
+			os.Exit(1)
+		}
+	}()
+
+	if err = srv.Serve(tcpl); err != nil {
```

Comment:
Ok, I'll do that. I was going to redo all this in a separate PR that also handles signals correctly but I can do some sync part here.